### ISSUE TITLE
Add S.Port / FBus

### ISSUE THREAD

**Issue by jasc76** [am32-firmware/AM32#23]
Hi there
can you add the possibility to output S.Port and FBus telemetry? It is a really nice thing to buy opensource esc's and have telemetry directly from the esc and AM32 sadly doesn't have this

**Comment by jasc76** [comment]
Any word on this feature?

**Comment by freasy** [comment]
This is not on our priority list. Keep in mind that we have limited flash.

**Comment by M-i-c-h-a-e-l1** [comment]
Maybe the following will help: [ESCape32](https://github.com/neoxic/ESCape32):

- Servo PWM, Oneshot125, automatic throttle calibration
- DSHOT 300/600/1200, bidirectional DSHOT, extended telemetry
- Analog/serial/iBUS/SBUS/SBUS2/CRSF input mode
**- KISS/iBUS/S.Port/CRSF telemetry**
- DSHOT 3D mode, turtle mode, beacon, LED, programming
- Sine startup mode (crawler mode), brushed mode
- Proportional brake, drag brake
- Temperature/voltage/current protection
- Variable PWM frequency, active freewheeling
- Customizable startup music

**Comment by AlkaMotors** [comment]
We have chosen to keep the number of auto detected input types at a minimum to reduce the chance of unwanted spoolup and misdirected signals. We will probably stick with Pwm, dshot and CAN, and crsf to keep it as safe as possible.

**Comment by jasc76** [comment]
Ok if you don’t add FBus, but S.Port telemetry would be great given the large userbase

**Comment by M-i-c-h-a-e-l1** [comment]
As my post to your reply on GitHub was deleted, maybe email works ...
Please have a look at ESCape32 on GitHub. This should have all the features
you are looking for.

Kind regards
M

Am Mi., 2. Okt. 2024 um 11:34 Uhr schrieb jasc76 ***@***.***>:

**Comment by freasy** [comment]
> Ok if you don’t add FBus, but S.Port telemetry would be great given the large userbase

We discussed it and agreed on adding s.port telemetry. Are you in the discord? I would appreciate it you could test it

**Comment by freasy** [comment]
@jasc76 on what ESC are you wanting this? Then i can send you a hex file to test.

**Comment by Satcomix** [comment]
Hello freasy,
I would like to help test the FrSky S.Port protocol.
I own several FrSky transmitters X20S, X20Pro, TandemXE and receivers up to 18 channels with S.Port or FBUS
I get several AM32 ESCs with voltage, current, capacity, temp, rpm sensors.
Sequre SQESC2670 70A G071
SQESC28120 120A G071
NeutronRC 70A G071
NeutronRC Exuav 70A F051
VIM Drones L431 AM32 development board with A1404 BL-Motor
VIM Drones L431 Drone CAN ESC Nano 15A.
I can also be found on Discord.
Can you send me please some .hex files for testing the S.Port protocol on my boards.
Best regards,
Torsten

**Comment by robthomson** [comment]
Happy to test too..

**Comment by jasc76** [comment]
> @jasc76 on what ESC are you wanting this? Then i can send you a hex file to test.

Oh sorry, late to the party 🙃
Sequre 28120

**Comment by Satcomix** [comment]
> > @jasc76 on what ESC are you wanting this? Then i can send you a hex file to test.
> 
> Oh sorry, late to the party 🙃 Sequre 28120
> 
> Sure, I can join your discord

Hello jasc76,
it`s never to late for S.Port telemetry :-))
https://discord.com/channels/1029170737342316605/1304092323646406719
see you,
greatings,
Torsten

**Comment by jasc76** [comment]
TBS discord !?

I won’t be able to best before the weekend. Will try it ✊🏻

**Comment by jasc76** [comment]
> > > @jasc76 on what ESC are you wanting this? Then i can send you a hex file to test.
> > 
> > 
> > Oh sorry, late to the party 🙃 Sequre 28120
> > Sure, I can join your discord
> 
> Hello jasc76, it`s never to late for S.Port telemetry :-)) https://discord.com/channels/1029170737342316605/1304092323646406719 see you, greatings, Torsten

Hmm
Can’t do anything with this discord link 🤷🏼‍♂️ its just the TBS general chat

**Comment by freasy** [comment]
> TBS discord !?
> 
> I won’t be able to best before the weekend. Will try it ✊🏻

no, am32 discord

**Comment by Satcomix** [comment]
> > > > @jasc76 on what ESC are you wanting this? Then i can send you a hex file to test.
> > > 
> > > 
> > > Oh sorry, late to the party 🙃 Sequre 28120
> > > Sure, I can join your discord
> > 
> > 
> > Hello jasc76, it`s never to late for S.Port telemetry :-)) https://discord.com/channels/1029170737342316605/1304092323646406719 see you, greatings, Torsten
> 
> Hmm Can’t do anything with this discord link 🤷🏼‍♂️ its just the TBS general chat

Hello jasc76,
Here is the discord link from the GitHub AM32 mainpage.
https://discord.gg/h7ddYMmEVV
Under #development there is the "S.Port" channel.
hope to see you there.

**Comment by jasc76** [comment]
For me the SPort from here works
https://github.com/pdingemans/AM32_SPORT/releases/tag/3.0.x

It would be really great if you could merge it !!!!!

**Comment by Satcomix** [comment]
> For me the SPort from here works https://github.com/pdingemans/AM32_SPORT/releases/tag/3.0.x
> 
> It would be really great if you could merge it !!!!!

Hello Jasc76, 
thank you very much for the tip about this very good news regarding the S.PORT protocol on the G071 chipset for AM32. 
I will test it extensively over the next few weeks. 
Best regards, Torsten

**Comment by M-i-c-h-a-e-l1** [comment]
On the Sequre 2670 make sure that a 100nF capacitor is on the board. Otherwise the current measurement will be completely off.

Kind regards
M.

**Comment by jasc76** [comment]
> On the Sequre 2670 make sure that a 100nF capacitor is on the board. Otherwise the current measurement will be completely off.
> 
> Kind regards M.

Do you have a pic where that cap is supposed to be !?

**Comment by jasc76** [comment]
Any news on this?
Will it be incorporated in the standard releases???

**Comment by jasc76** [comment]
please please please .....

Its been two years and there is already code available. 
And Escape32 now implements CRSF and you can program directly with the TX....

**Comment by robthomson** [comment]
My view is simple enough. 

Zero interest in this code from devs.   I will just fork and make a version that works.. Tracking this project as the upstream.

**Comment by freasy** [comment]
My self for example lack the hardware to test this. I will try to get it. For CRSF, I have the code ready, we must clear some license questions and then will add it too.

**Comment by robthomson** [comment]
Fbus support will not be possible without a license fee.  Sport is pretty simple. 

On my pr there is one change I think I would like to make that will make config from the radio easier long term.

**Comment by AlkaMotors** [comment]
I tested the builds to see if its a possibility to add to the config tool but we are runnig low on flash space on some mcus. I don't have a problem adding this as a build option.  I was just trying to work it in where its easier to turn on and off. Alternately if you do host a fork I will link to it from the Readme.

**Comment by freasy** [comment]
I just talked to FrSky and will get hardware as soon as possible. Also we got the permission to implement F.Port 🥳

**Comment by robthomson** [comment]
Who did you speak too? 

I can likely ease any work you need in this regard.

**Comment by jasc76** [comment]
> I just talked to FrSky and will get hardware as soon as possible. Also we got the permission to implement F.Port 🥳

Oh, thats great !! F.Port or FBus also ??

**Comment by freasy** [comment]
I am not sure if i can share that, i also have to sign an NDA.

I think there is no such thing as F.Port (my mistake), its F.Bus and its what we have permission to.

**Comment by robthomson** [comment]
Absolutely.   I have all the docs myself.   Whats your discord id?   I am on the am32 as rob.thomson

**Comment by freasy** [comment]
I am wrong alltogether :D there seem to be both? I will clarify this.

**Comment by Satcomix** [comment]
> I just talked to FrSky and will get hardware as soon as possible. Also we got the permission to implement F.Port 🥳

Hello Eike,
if you need FrSky hardware as TX (X20S, XE), RX (ArcherPlus Series) and different Sensors i can send it to you. It`s no so a far way from NE :-))
Greetings,
Torsten

**Comment by jasc76** [comment]
> I am not sure if i can share that, i also have to sign an NDA.
> 
> I think there is no such thing as F.Port (my mistake), its F.Bus and its what we have permission to.

FBus is fine. Its the new standard. S.Port and FBus are giving the best combination imho

**Comment by jasc76** [comment]
Hi
Any movement on this topic?
